### PR TITLE
[IMP] mail: introduce ChatWindow/onClickClose

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2579306

### CODE REVIEW

File: addons/mail/static/src/models/chat_window.js [hunk 1]
```diff
@@ -104,6 +104,16 @@ registerModel({
             const lastVisible = this.manager.lastVisible;
             this.manager.swap(this, lastVisible);
         },
+        /**
+         * @param {MouseEvent} ev
+         */
+        onClickClose(ev) {
+            ev.stopPropagation();
+            if (!this.exists()) {
```
Comment (phenix-factory) [inline addons/mail/static/src/models/chat_window.js:112]: Do we really need this inside a `onClick` inside a model ?

Comment (alexkuhn) [reply addons/mail/static/src/models/chat_window.js:112]: Yes, we cannot guarantee that invoking `t-on` in OWL components only happens on existing records.
It's possible that record is itself deleted, so we need to guard them.

Comment (phenix-factory) [reply addons/mail/static/src/models/chat_window.js:112]: In my understanding, it's only valid for even that have a chance to trigger this guard (e.g. on-scroll)
If not, we need to do it consistently on every `onClick` (https://github.com/odoo/odoo/pull/88231/files#diff-564a4336843bb910f681be30591a6113725c8c61c802683e65833cdbc2779109R107).

Comment (alexkuhn) [reply addons/mail/static/src/models/chat_window.js:112]: Yes, I think all `t-on` are affected by the problem. The ones that are not guarded do work currently, but are prone to break with small changes to the code.

Comment (seb-odoo) [reply addons/mail/static/src/models/chat_window.js:112]: All of them potentially have the issue, but the probability of the issue on click is so low (clicking exactly before the animation frame of the delete) that we never really bothered with them, and we've never seen or heard about the issue on them in 2  years.

Comment (seb-odoo) [reply addons/mail/static/src/models/chat_window.js:112]: I would not introduce extra guards on handlers, but I would keep those that we have (it's likely that the issue was manifesting on them to make us add it at some point).

**Review by phenix-factory — APPROVED** [review]
LGTM

File: addons/mail/static/src/models/chat_window.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/models/chat_window.js:112]: I don't mind introducing a "handler" category to have guard on all of them.
Just for short/mid term, to hide technical problem and have ease on mind for at least this problem until we have overall improved code that won't need this eventually. It's only a problem due to integration with OWL components and lack of maturity to model.js code.

Personally I understand we need the guard, but it's annoying to make changes and have a test failing and reminding me of this subtle technical detail that we have to deal with.

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/88468

**Pull request closed without merging**